bluemira/power_cycle/net.py

```diff
@@ -0,0 +1,660 @@
+        unit:
+            return unit, defaults to [W]
+        end_time:
+            for unnormalised loads this assures the load is
+            applied at the right point in time
+        """
+        timeseries, end_time = self._normalise_timeseries(timeseries, end_time)
+        load_type = LoadType.from_str(load_type)
+        return {
+            load.name: load.interpolate(timeseries, end_time, load_type)
+            if unit is None
+            else raw_uc(
+                load.interpolate(timeseries, end_time, load_type),
+                load.unit,
+                unit,
+            )
+            for load in self.loads.values()
+        }
+
+    def load_total(
+        self,
+        timeseries: np.ndarray,
+        load_type: Union[str, LoadType],
+        unit: Optional[str] = None,
+        end_time: Optional[float] = None,
+    ) -> np.ndarray:
+        """Total load for each timeseries point for a given load_type
+
+        Parameters
+        ----------
+        timeseries:
+            time array
+        load_type:
+            Type of load
+        unit:
+            return unit, defaults to [W]
+        end_time:
+            for unnormalised loads this assures the load is
+            applied at the right point in time
+        """
+        return np.sum(
+            list(
+                self.get_load_data_with_efficiencies(
+                    timeseries, load_type, unit, end_time
+                ).values()
+            ),
+            axis=0,
+        )
+
+
+@dataclass
+class Phase:
+    """Phase container"""
+
+    config: PhaseConfig
+    subphases: Dict[str, PowerCycleSubPhase]
+    loads: Loads
+
+    def __post_init__(self):
+        """Validate duration"""
+        if self.duration < 0:
+            raise ValueError(
+                f"{self.config.name} phase duration must be positive: {self.duration}s"
+            )
+
+    @property
+    def duration(self):
+        """Duration of phase"""
+        data = [s_ph.duration for s_ph in self.subphases.values()]
+        try:
+            return getattr(np, self.config.operation)(data)
+        except np.core._exceptions.UFuncTypeError as e:
+            raise TypeError(f"duration variables have not been imported {data}") from e
+
+
+@dataclass
+class PowerCycleLibraryConfig:
+    """Power Cycle Configuration"""
+
+    scenario: ScenarioConfigDescriptor = ScenarioConfigDescriptor()
+    pulse: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PulseConfig)
+    phase: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PhaseConfig)
+    subphase: LibraryConfigDescriptor = LibraryConfigDescriptor(
+        config=PowerCycleSubPhase
+    )
+    system: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PowerCycleSystem)
+    subsystem: LibraryConfigDescriptor = LibraryConfigDescriptor(
+        config=PowerCycleSubSystem
+    )
+    loads: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PowerCycleLoadConfig)
+
```

Comment:
```suggestion
class PowerCycleLibraryConfig:
    """Power Cycle Configuration"""

    def __init__(
        self,
        scenario: ScenarioConfig,
        pulse: Dict[str, PulseConfig],
        phase: Dict[str, PhaseConfig],
        subphase: Dict[str, PowerCycleSubPhase],
        system: Dict[str, PowerCycleSystem],
        subsystem: Dict[str, PowerCycleSubSystem],
        loads: Dict[str, PowerCycleLoadConfig],
    ):
        self.scenario = scenario
        self.pulse = pulse
        self.phase = phase
        self.subphase = subphase
        self.system = system
        self.subsystem = subsystem
        self.loads = loads

        self.check_config()
```

Reply:
I haven't done the check config at the end of the init because I can see a situation where eg the load hasnt been added yet. The check is still done on the get_* functions